modules/core/02-client/keeper/keeper.go

```diff
@@ -241,27 +241,27 @@ func (k Keeper) GetLatestClientConsensusState(ctx sdk.Context, clientID string)
 // GetSelfConsensusState introspects the (self) past historical info at a given height
 // and returns the expected consensus state at that height.
 // For now, can only retrieve self consensus states for the current revision
-func (k Keeper) GetSelfConsensusState(ctx sdk.Context, height exported.Height) (exported.ConsensusState, bool) {
+func (k Keeper) GetSelfConsensusState(ctx sdk.Context, height exported.Height) (exported.ConsensusState, error) {
 	selfHeight, ok := height.(types.Height)
 	if !ok {
-		return nil, false
+		return nil, sdkerrors.ErrInvalidType.Wrapf("unable cast %T to %T", height, types.Height{})
```

Comment:
```suggestion
		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "expected %T, got %T", types.Height{}, height)
```
one last nit, thanks!